public/modules/custom/helfi_ahjo/src/Form/AhjoConfigForm.php

```diff
@@ -0,0 +1,285 @@
+      '#title' => $this->t('Ahjo Base URL'),
+      '#default_value' => $config->get('base_url'),
+      '#description' => $this->t('example: demo.ahjo.fi'),
+      '#required' => TRUE,
+    ];
+
+    $form['api_config']['helfi_ahjo_api_key'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Ahjo API Key'),
+      '#default_value' => $config->get('api_key'),
+      '#description' => $this->t('apikey'),
+      '#required' => TRUE,
+    ];
+
+    $form['cron_config'] = [
+      '#type' => 'fieldset',
+      '#title' => $this
+        ->t('Cron Configs'),
+    ];
+    $form['cron_config']['sync_interval'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Ahjo Sections Update Interval'),
+      '#options' => [
+        '-1' => $this->t('Every cron run'),
+        '3600' => $this->t('Every hour'),
+        '7200' => $this->t('Every 2 hours'),
+        '10800' => $this->t('Every 3 hours'),
+        '14400' => $this->t('Every 4 hours'),
+        '21600' => $this->t('Every 6 hours'),
+        '28800' => $this->t('Every 8 hours'),
+        '43200' => $this->t('Every 12 hours'),
+        '86400' => $this->t('Every 24 hours'),
+      ],
+      '#default_value' => empty($config->get('sync_interval')) ? 86400 : $config->get('sync_interval'),
+      '#description' => $this->t('How often should Ahjo Sections be synced with AhjoProxy?'),
+      '#required' => TRUE,
+    ];
+
+    $form['cron_config']['org_id'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Organisation ID (Start)'),
+      '#default_value' => $config->get('org_id') ?? 00001,
+      '#description' => $this->t('example: 00001'),
+      '#required' => TRUE,
+    ];
+
+    $form['cron_config']['max_depth'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Max Depth'),
+      '#default_value' => $config->get('max_depth') ?? 9999,
+      '#description' => $this->t('example: 9999'),
+      '#required' => TRUE,
+    ];
+
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save Ahjo Configuration'),
+      '#button_type' => 'primary',
+    ];
+
+    $form['cron_config']['actions']['import_sync'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Sync now'),
+      '#button_type' => 'primary',
+      '#submit' => ['::importSyncData'],
+    ];
+
+    $form['actions']['deleteAllData'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Delete all imported data'),
+      '#button_type' => 'primary',
+      '#submit' => ['::deleteAllData'],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Validate that the provided values are valid or nor.
+   *
+   * @param array $form
+   *   Form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state instance.
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    $base_url = Xss::filter($form_state->getValue('helfi_ahjo_base_url'));
```

Comment:
these xss filter are not needed. in general, the xss filtering should be done at rendering, not saving